pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -47,13 +49,31 @@ def __init__(self) -> None:
         self.running_loss: TensorRunningAccum = TensorRunningAccum(window_length=20)
         self.batch_idx: int = 0
         self.split_idx: Optional[int] = None
-        self.warning_cache: WarningCache = WarningCache()
 
+        self.progress = BatchProgress()
+        self.optim_progress = OptimizationProgress()
+
+        self._warning_cache: WarningCache = WarningCache()
         self._hiddens: Optional[Tensor] = None
         self._optimizer_freq_cumsum: Optional[int] = None
         self._remaining_splits: Optional[List[Any]] = None
         self._skip_backward: bool = False
 
+    def connect(
+        self,
+        trainer: 'pl.Trainer',
+        *args: Any,
+        progress: Optional[BatchProgress] = None,
+        optim_progress: Optional[OptimizationProgress] = None,
+        **kwargs: Any
+    ) -> None:
+        """Connects the loop with necessary arguments like the trainer"""
+        super().connect(trainer, *args, **kwargs)
+        if progress is not None:
+            self.progress = progress
+        if optim_progress:
+            self.optim_progress = optim_progress
+
```

Comment:
could you describe why progress/optim_progress are passed through connect?

Reply:
We need to connect the progress instances between loops, just as we do for the trainer instance.

This is what is checked in the test added